ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1180 @@
+         *
+         * @type      {boolean}
+         * @since     next
+         */
+        zIndex: 7
+    };
+
+    /* *
+     *
+     * Properties
+     *
+     * */
+
+    public breadcrumbsGroup: SVGElement = void 0 as any;
+    public breadcrumbsList: Array<Array<any>> = [];
+    public chart: Chart;
+    public isDirty: boolean = true;
+    public level: number = -1;
+    public options: Breadcrumbs.BreadcrumbsOptions = void 0 as any;
+    public series?: TreemapSeries = void 0 as any;
+
+    public constructor(chart: Chart, userOptions?: Partial<Breadcrumbs.BreadcrumbsOptions>) {
+        const chartOptions = merge(
+            chart.options.drilldown &&
+                chart.options.drilldown.drillUpButton,
+            Breadcrumbs.defaultBreadcrumbsOptions,
+            chart.userOptions.drilldown &&
+            chart.userOptions.drilldown.drillUpButton,
+            userOptions
+        );
+
+        this.chart = chart;
+        this.options = chartOptions || {};
+    }
+
+    /**
+     * Calcule level on which chart currently is.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#calculateLevel
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     */
+    public calculateLevel(this: Breadcrumbs): void {
+        const chart = this.chart,
+            drilldownLevels = chart.drilldownLevels;
+
+        // Calculate on which level we are now.
+        this.level = drilldownLevels && drilldownLevels[drilldownLevels.length - 1] &&
+        drilldownLevels[drilldownLevels.length - 1].levelNumber + 1 || 0;
+    }
+
+    /**
+     * This method creates an array of arrays containing a level number
+     * with the corresponding series/point.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#createList
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     * @return {any}
+     *        Formatted text.
+     */
+    public createList(this: Breadcrumbs): any {
+        const breadcrumbsList: Array<Array<number|null|Point|SeriesType|SVGAElement>> = this.breadcrumbsList || [],
+            chart = this.chart,
+            drilldownLevels = chart.drilldownLevels;
+
+        // The list is based on drilldown levels from the chart object
+        if (drilldownLevels && drilldownLevels.length) {
+            // Add the initial series as the first element.
+            if (!breadcrumbsList[0]) {
+                breadcrumbsList.push([0, (drilldownLevels[0].seriesOptions as Point)]);
+            }
+
+            const lastBreadcrumb = breadcrumbsList[breadcrumbsList.length - 1];
+
+            drilldownLevels.forEach(function (level): void {
+                // If level is already added to breadcrumbs list,
+                // don't add it again- drilling categories
+                // + 1 because of the wrong levels numeration
+                // in drilldownLevels array.
```

Comment:
Q: Why it's wrong?

Reply:
The initial main level is not added, every next is added with increased index, starting from 0. That means that the first drilldown level is actually with index 0.